src/App.php

```diff
@@ -290,7 +291,10 @@ public function dispatch()
 
             $result = call_user_func_array(
                 $routeParams['_controller'],
-                array($this->serviceManager->get('Request'))
+                array(
+                    $this->serviceManager->get('Request'),
+                    $this->serviceManager->get('Response'),
```

Comment:
Closer to the middleware-compatible pattern.

If you don't then how else are you going to craft a response from controller. You could do $this->getService('Request'), but that's not as favourable as dependency injection.